zstacklib/zstacklib/network/ovs/dpdk.py

```
@@ -0,0 +1,626 @@
+        except OSError as err:
+            raise OvsError(str(err))
+
+    def _list_ifaces_by_vm_uuid(self, vm_uuid: str) -> list[str]:
+        """List interfaces by VM UUID."""
+        try:
+            interfaces = shell.call(
+                CTL_BIN + f"--columns=name find interface external_ids:vm-id={vm_uuid} | grep name | cut -d ':' -f2 | tr -d ' '"
+            )
+            return interfaces.strip().splitlines()
+        except shell.ShellError as err:
+            raise OvsError(f'Find interface by vm uuid:{vm_uuid} failed. {err}')
```

Ruff (0.15.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)